docs/protocol.md

```diff
@@ -0,0 +1,333 @@
+  | *Block-specific fields* | |                                           |
+  | Signature | 64           | Ed25519 signature of the block's hashables |
+  | Work      | 8            | Unsigned 64-bit integer (little endian?)   |
+
+The "hashables" of a block are the set of field values that are fed into
+the Blake2b hashing algorithm in order to get a hash value for the block.
+This hash value is then signed with the public key of the account that generated the block
+and stored in the Signature field.
+
+XXX Work value
+
+The different block types are denoted by a Type value (see `enum class block_type` in `rai/lib/blocks.hpp`):
+
+  | Block type | Value |
+  | ---------- | ----- |
+  | Send       | 2     |
+  | Receive    | 3     |
+  | Open       | 4     |
+  | Change     | 5     |
+  | State      | 6     |
+
+## Transfer of funds
+
+For each block in an account's chain a balance value can be determined, either from
+a value stored in the block itself or by using differences between successive
+blocks in the chain. Differences in balance values between successive blocks
+can be used to compute the amount of funds transferred by blocks to other
+accounts.
+
+Only Send and State blocks contain an explicit Balance value. For other block
+types balances and amounts needs to be deducted recursively:
+
+* The balance for an Open block is equal to the amount being sent by the corresponding
+  Send block. An exception to this is the Open block for the Genesis account, which always
+  receives the maximum supply of 2<sup>128</sup>-1 raw.
+* The balance for a Receive block is the balance of its predecessor in the
+  account chain plus the amount sent by the corresponding Send block.
+* The amount of funds being sent by a Send block is defined implicitly as
+  the difference between the balance at the previous block in the account chain
+  and the balance at the Send block. This amount must be > 0.
+* A Change block has no influence on an account's balance, so its balance is
+  equal to the balance of its predecessor.
+
+All State blocks contain a Balance value, hence determining amounts sent or
+received from one State block to the next is straightforward.
+
+## Send block
+
+Reference: `rai::send_block` in `rai/lib/blocks.cpp`
+
+A Send block transfers funds from the account the block belongs to to a
+different account.
+
+A Send block contains the following block-specific fields:
+
+  | Field         | Size (bytes) | Description |
+  | ------------- | ----- | ----------------------------------------------- |
+  | Previous      | 32    | Hash of the previous block in the account chain |
+  | Destination   | 32    | The destination account being sent to |
+  | Balance       | 32    | The sending account's new balance |
+
+Note that the Balance value in the block is the balance of the sending
+account *after the amount sent in the block has been deducted*.
+
+XXX is it valid to send to yourself?
```

Comment:
Yes, with the old "send" block type, you can send to yourself, even 0 value transactions.

Reply:
You can also send to yourself with the new state blocks, unless I forgot about something.